### DISCUSSION TITLE
pattern matching problem with whitespace

### DISCUSSION THREAD

**Discussion by thk12** [RsyncProject/rsync#352] [Q&A]
hello there,

surely not a bug, but: I'am using rsync v3.2.3 and have a problem I don't understand with the following pattern:

```
pi:~ $ find source/
source/
source/hello
source/hello/whitespace_dir
source/hello/whitespace_file
source/foo
source/foo/bar1
source/foo/bar2
```

rsync pattern example:

```
pi:~ $ rsync -aHvv --dry-run --include="*/" --include="bar" --exclude="*" source/ target/
sending incremental file list
[sender] showing directory hello because of pattern */
[sender] showing directory foo because of pattern */
[sender] showing directory foo/bar1 because of pattern */
[sender] showing directory foo/bar2 because of pattern */
[sender] showing directory hello/whitespace_dir  because of pattern */
[sender] hiding file hello/whitespace_file  because of pattern *
delta-transmission disabled for local transfer or --whole-file
./
foo/
foo/bar1/
foo/bar2/
hello/
hello/whitespace_dir /
total: matches=0  hash_hits=0  false_alarms=0 data=0
```

I just dont get it, why "hello/whitespace_dir " matches (and "source/hello/whitespace_file " not?) Any suggests?

greetings, t

**Comment by WayneD** [comment]
Your include rules only include all directories (`*/`) and a file named `bar` and nothing else.  Everything else is excluded by your trailing rule. If you want the whilespace file to be included, you need a rule (prior to the exclude) to include it.

**Reply by thk12** [reply]
thanks a lot for explanation, but I dont want do include the whitespace file, rather I want to exclude the whitespace directory. With other words: I want to rsync only objects matching ```bar``` (```recursively```) and nothing else.

**Comment by WayneD** [comment]
There is nothing named bar in your list of files.  Maybe you want to include `bar*` for all things beginning with "bar". However, you have to remember to include any path elements on the way down to the bar dirs.  For instance:

`--include='bar*' --include='/foo/' --exclude='*'`

That would include the dirs without their contents.  If you want everything inside the dirs too, you could prefix a `--include='bar*/**'` rule.  See the newly enhanced [filter rules section of the manpage](https://download.samba.org/pub/rsync/rsync.1#FILTER_RULES) for more info.
